lib/trino-hive-formats/src/main/java/io/trino/hive/formats/compression/ChunkedSliceOutput.java

```diff
@@ -1,386 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package io.trino.hive.formats.compression;
-
-import com.google.common.collect.ImmutableList;
-import io.airlift.slice.Slice;
-import io.airlift.slice.SliceOutput;
-import io.airlift.slice.Slices;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.Charset;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-
-import static com.google.common.base.Preconditions.checkArgument;
-import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
-import static io.airlift.slice.SizeOf.SIZE_OF_INT;
-import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
-import static io.airlift.slice.SizeOf.SIZE_OF_SHORT;
-import static io.airlift.slice.SizeOf.instanceSize;
-import static java.lang.Math.min;
-import static java.lang.Math.multiplyExact;
-import static java.lang.Math.toIntExact;
-
-public final class ChunkedSliceOutput
```

Comment:
Why not delete the one in orc instead ?
Having it here makes it easier for other file formats to use this as well

Reply:
please see where `trino-orc` is used and where `trino-hive-formats` is used. do you still think we should do other way?

Reply:
Okay, we shouldn't take dependency of `trino-hive-formats` in `trino-orc` as it gets included in iceberg and delta. But now I also think taking dependency of `trino-orc` in `trino-hive-formats` is confusing as it looks like `trino-hive-formats` is supposed to be code for hive file formats other than orc and parquet.
Maybe we move this class to `trino-plugin-toolkit` ?

Reply:
awesome idea

Reply:
done, ptal